File: sekolahunggulintegrasi/backend/sekolah/models/Proposal.js
const mongoose = require('mongoose');
const proposalSchema = new mongoose.Schema({
  jenisProposal: {
    type: String,
    required: true,
    enum: [
      'Renovasi Bangunan',
      'Pembangunan Baru',
      'Penambahan Fasilitas',
      'Perbaikan Infrastruktur Penunjang (Jalan, Pagar, Toilet, Dll)'
    ],
  },
  judulProposal: {
    type: String,
    required: true,
  },
  namaSekolah: {
    type: String,
    required: true,
  },
  alamatSekolah: {
    type: String,
    required: true,
  },
  jumlahSiswa: {
    type: Number,
    required: true,
  },
  fasilitas: {
    type: String,
    required: true,
  },
  dokumenProposalPath: {
    type: String,
    required: true,
  },
  desainRencanaPath: {
    type: String,
    required: true,
  },
  dokumenRABPath: {
    type: String,
    required: true,
  },
  status: {
    type: String,
    default: 'Menunggu',
    enum: ['Menunggu', 'Disetujui', 'Ditolak'],
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Proposal', proposalSchema);

File: sekolahunggulintegrasi/lib/models/proposal.dart
// lib/models/proposal.dart
class Proposal {
  final String id;
  final String jenisProposal;
  final String judulProposal;
  final String namaSekolah;
  final String alamatSekolah;
  final int jumlahSiswa;
  final String fasilitas;
  final String dokumenProposalPath;
  final String desainRencanaPath;
  final String dokumenRABPath;
  final String status;
  final DateTime createdAt;

  Proposal({
    required this.id,
    required this.jenisProposal,
    required this.judulProposal,
    required this.namaSekolah,
    required this.alamatSekolah,
    required this.jumlahSiswa,
    required this.fasilitas,
    required this.dokumenProposalPath,
    required this.desainRencanaPath,
    required this.dokumenRABPath,
    required this.status,
    required this.createdAt,
  });

  factory Proposal.fromJson(Map<String, dynamic> json) {
    return Proposal(
      id: json['_id'],
      jenisProposal: json['jenisProposal'],
      judulProposal: json['judulProposal'],
      namaSekolah: json['namaSekolah'],
      alamatSekolah: json['alamatSekolah'],
      jumlahSiswa: json['jumlahSiswa'],
      fasilitas: json['fasilitas'],
      dokumenProposalPath: json['dokumenProposalPath'],
      desainRencanaPath: json['desainRencanaPath'],
      dokumenRABPath: json['dokumenRABPath'],
      status: json['status'] ?? 'Menunggu',
      createdAt: DateTime.parse(json['createdAt'] ?? DateTime.now().toIso8601String()),
    );
  }
}